crates/web5/src/errors.rs

```diff
@@ -10,6 +13,16 @@ pub enum Web5Error {
     DataMember(String),
     #[error("not found error {0}")]
     NotFound(String),
+    #[error("cryptography error {0}")]
+    Crypto(String),
+    #[error("encoding error {0}")]
+    Encoding(String),
+}
+
+impl From<DecodeError> for Web5Error {
+    fn from(err: DecodeError) -> Self {
+        Web5Error::Encoding(err.to_string())
+    }
```

Comment:
I remember we chatted about this at standup but I had to think it through again. Writing the train of thought here in case anyone else has the same question:

why manually implement `From<>` for error conversions instead of using `#[from]` from `thiserror`? Because `#[from]` puts the the source error as the associated payload -- e.g. `Web5Error::Encoding(base64::DecodeError)`. Whereas we want to use a `String` as the associated payload because this displays better in error stack traces in bound languages.

Reply:
The matter we discussed was actually about the `serde_json::Error` which we were forced to implement the `From` trait for because it doesn't derive the `Clone`, but it's a good question because it's brought me to the _real_ reason I think we should go the `From<>` route instead of the `#[from]` route, which is encapsulation/coupling. With `#[error(transparent)]Encoding(#[from] DecodeError)` we do indeed have a richer set of information and stronger types, however, we leak `base64` abstractions across our publicly accessible API. Or said differently, we're strongly coupling our crate to `base64` as well as the consumers of our crate. But `base64` is an encapsulated detail, and if it's not and we choose to change to a different dependency in the future then we would be introducing a breaking change. So like, I would be in favor of the `#[from]` route for errors which are strictly encapsulated, but anything which is exposed to the publicly accessible API should be our own types. 

So basically, TL;DR prevent abstraction bleed.